lstchain/reco/reconstructor.py

```diff
@@ -0,0 +1,977 @@
+    sigma_space = Float(4, help='', allow_none=False).tag(config=True)
+    sigma_time = Float(3, help='', allow_none=False).tag(config=True)
+    time_before_shower = Float(10, help='', allow_none=False).tag(config=True)
+    time_after_shower = Float(20, help='', allow_none=False).tag(config=True)
+    use_weight = Bool(False, help='', allow_none=False).tag(config=True)
+    no_asymmetry = Bool(False, help='', allow_none=False).tag(config=True)
+    use_interleaved = Path(None, help='', allow_none=True).tag(config=True)
+    telescope_id = Int(1, help='', allow_none=False).tag(config=True)
+    # Allows only one telescope for now, need to use telescope-wise information later
+    n_peaks = Int(50, help='', allow_none=False).tag(config=True)
+    verbose = Int(0, help='', allow_none=False).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+        self.geometry = self.subarray.tel[self.telescope_id].camera.geometry
+        self.pix_x = self.geometry.pix_x.to_value(u.m)
+        self.pix_y = self.geometry.pix_y.to_value(u.m)
+        self.pix_area = self.geometry.pix_area.to(u.m ** 2).value
+        self.r_max = np.sqrt(self.pix_x ** 2 + self.pix_y ** 2).max()
+        self.focal_length = subarray.tel[self.telescope_id].optics.equivalent_focal_length
+        readout = self.subarray.tel[self.telescope_id].camera.readout
+        sampling_rate = readout.sampling_rate.to_value(u.GHz)
+        self.dt = (1.0 / sampling_rate)
+        self.labels = {'charge': 'Charge [p.e.]',
+                       't_cm': '$t_{CM}$ [ns]',
+                       'x_cm': '$x_{CM}$ [m]',
+                       'y_cm': '$y_{CM}$ [m]',
+                       'length': r'$\sigma_l$ [m]',
+                       'wl': r'$\sigma_w$ / $\sigma_l$',
+                       'psi': r'$\psi$ [rad]',
+                       'v': '$v$ [m/ns]',
+                       'rl': 'length asymmetry'
+                       }
+        self.names_parameters = list(inspect.signature(self.log_pdf).parameters)
+        self.template = NormalizedPulseTemplate.load_from_eventsource(subarray.tel[1].camera.readout)
+        self.template_time_of_max = self.template.compute_time_of_max()
+        if self.use_interleaved is None:  # test to include interleaved correction on pedestal, NOT FUNCTIONAL
+            self.pedestal_std = None
+        else:
+            _, self.pedestal_std = get_bias_and_std(config['lh_fit_config']['use_interleaved'])
+        self.end_parameters = None
+        self.error_parameters = None
+        self.fcn = None
+
+        poisson_peaks = np.arange(self.n_peaks, dtype=int)
+        poisson_peaks[0] = 1
+        log_factorial = np.log(poisson_peaks)
+        self.log_factorial = np.cumsum(log_factorial)
+
+    def __call__(self, event, dl1_container):
+        self.event_id = event.index.event_id
```

Comment:
Why are these stored on the class? After `__call__` is over, they should not be needed anymore.

Reply:
It is actually not used in __call__ but only for the plotting function to save the figure.
As we discussed before, we don't save the information needed for this plots and they are very useful to identify issues in development. So I would like to keep them for now.
One possibility I though about would be to import them from a dedicated file taking the fitter as input.

Now that I think about it, passing a string to TimeWaveformFitter.plot_debug() instead of always using the event id could also work. And this would remove the need to save it in the fitter.